src/__swaps__/screens/Swap/components/SwapActionButton.tsx

```diff
@@ -0,0 +1,176 @@
+import { AnimatedText, Box, Column, Columns, Text, globalColors, useColorMode, useForegroundColor } from '@/design-system';
+import { useTheme } from '@/theme';
+
+export const SwapActionButton = ({
+  color,
+  borderRadius,
+  disableShadow,
+  hugContent,
+  icon,
+  iconStyle,
+  label,
+  onLongPress,
+  onPress,
+  outline,
+  rightIcon,
+  scaleTo,
+  small,
+}: {
+  color?: string;
+  borderRadius?: number;
+  disableShadow?: boolean;
+  hugContent?: boolean;
+  icon?: string | Readonly<SharedValue<string | undefined>>;
+  iconStyle?: StyleProp<TextStyle>;
+  label: string | Readonly<SharedValue<string | undefined>>;
+  onLongPress?: () => void;
+  onPress?: () => void;
+  outline?: boolean;
+  rightIcon?: string;
+  scaleTo?: number;
+  small?: boolean;
+}) => {
+  const { isDarkMode } = useColorMode();
+  const { colors } = useTheme();
+
+  const fallbackColor = useForegroundColor('blue');
+  const separatorSecondary = useForegroundColor('separatorSecondary');
+
+  const textColor = useMemo(() => {
+    if (!color) return globalColors.white100;
+    const contrastWithWhite = c.contrast(color, globalColors.white100);
+
+    if (contrastWithWhite < (isDarkMode ? 2.6 : 2)) {
+      return globalColors.grey100;
+    } else {
+      return globalColors.white100;
+    }
+  }, [color, isDarkMode]);
+
+  const secondaryTextColor = useMemo(() => {
+    if (!color) return colors.alpha(globalColors.white100, 0.76);
+    const contrastWithWhite = c.contrast(color, globalColors.white100);
+
+    if (contrastWithWhite < (isDarkMode ? 2.6 : 2)) {
+      return colors.alpha(globalColors.grey100, 0.76);
+    } else {
+      return colors.alpha(globalColors.white100, isDarkMode ? 0.76 : 0.8);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [color, isDarkMode]);
+
+  return (
+    <ButtonPressAnimation
+      onLongPress={onLongPress}
+      onPress={onPress}
+      scaleTo={scaleTo || (hugContent ? undefined : 0.925)}
+      style={hugContent && feedActionButtonStyles.buttonWrapper}
+    >
+      <Box
+        paddingHorizontal={{ custom: small ? 14 : 20 - (outline ? 2 : 0) }}
+        paddingLeft={small && icon ? '10px' : undefined}
+        paddingRight={small && rightIcon ? '10px' : undefined}
+        style={[
+          feedActionButtonStyles.button,
+          outline && feedActionButtonStyles.outlineButton,
+          {
+            backgroundColor: outline ? 'transparent' : color || fallbackColor,
+            borderColor: outline ? separatorSecondary : undefined,
+            borderRadius: borderRadius ?? 24,
+            height: small ? 36 : 48,
+            shadowColor: disableShadow || outline ? 'transparent' : color || fallbackColor,
+            shadowOffset: {
+              width: 0,
+              height: isDarkMode ? 13 : small ? 6 : 10,
+            },
+            shadowOpacity: isDarkMode ? 0.2 : small ? 0.2 : 0.36,
+            shadowRadius: isDarkMode ? 26 : small ? 9 : 15,
+          },
+        ]}
```

Comment:
out of scope for this obviously but i would love for this to become a styled component that receives all these props & we can extract this out from a component's return statement